hibernate-core/src/main/java/org/hibernate/query/sql/spi/ParameterOccurrence.java

```diff
@@ -9,5 +9,5 @@
 /**
  * @author Christian Beikov
  */
-public record ParameterOccurrence(QueryParameterImplementor<?> parameter, int sourcePosition) {
+public record ParameterOccurrence(QueryParameterImplementor<?> parameter, int sourcePosition, int length) {
```

Comment:
I guess we need to add this to avoid breaking this SPI.

```suggestion
public record ParameterOccurrence(QueryParameterImplementor<?> parameter, int sourcePosition, int length) {
    @Deprecated(forRemoval = true)
    public ParameterOccurrence(QueryParameterImplementor<?> parameter, int sourcePosition) {
        this( parameter, sourcePosition, 1 );
    }
```